mixer/v1/config/client/auth.proto

```diff
@@ -0,0 +1,148 @@
+// Copyright 2017 Istio Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+syntax = "proto3";
+
+package istio.mixer.v1.config.client;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+message AuthPolicy {
+  // APIKey defines configuration for generating the API key attribute
+  // from HTTP requests.
+  //
+  // See https://swagger.io/docs/specification/authentication/api-keys
+  // for a general overview of API keys as defined by OpenAPI.
+  message APIKey {
+    oneof key {
+      // API Key is sent as a query parameter. `query` represents the
+      // query string parameter name.
+      //
+      // For example, `query=api_key` should be used with the
+      // following request:
+      //
+      //     GET /something?api_key=abcdef12345
+      //
+      string query = 1;
+      // API key is send in a request header. `header` represents the
+      // header name.
+      //
+      // For example, `header=X-API-KEY` should be used with the
+      // following request:
+      //
+      //     GET /something HTTP/1.1
+      //     X-API-Key: abcdef12345
+      //
+      string header = 2;
+
+      // API key is sent in a
+      // [cookie](https://swagger.io/docs/specification/authentication/cookie-authentication),
+      //
+      // For example, `cookie=X-API-KEY` should be used for the
+      // following request:
+      //
+      //     GET /something HTTP/1.1
+      //     Cookie: X-API-KEY=abcdef12345
+      //
+      string cookie = 3;
+    }
+
+    // Attribute name to use when forwarding api_key to mixer,
+    // e.g. api.key
+    string attribute_name = 4;
+  }
+
+  message JWT {
+    // TODO - does filter need a hint for where to extract JWT token
+    // from, e.g. URL, query parameter, or header?
+
+    // Value to insert into `sec-istio-auth-userinfo` header (or
+    // attribute?)
+    enum UserInfoType {
+      PAYLOAD = 0; // Payload JSON
+      PAYLOAD_BASE64_URL = 1; // base64url-encoded Payload JSON
+      HEADER_PAYLOAD_BASE64_URL = 2; // JWT with signature
+    }
+    UserInfoType user_infoType;
```

Comment:
This specifies how to pass the token claim payload to backend.   It should be global per service

Reply:
I think we can just decide one way to go, and also allow configuration forward original one or not is enough.

Reply:
I was about to propose:  Just like ESP,  we always forward it.   It is not a lot of data.  
But for data security, we may not want to forward it.    But again,  Istio backend is always localhost for side-car.  
For ingress proxy,  it is client proxy, backend is server proxy.  But Istio has mTL.  so it is also secure.

Reply:
So something like the following? We could also omit initially with default behavior of `ALL` and then add `NONE` later if there is a need.

```proto
enum UserInfoType {
    ALL = 0;  // forward the entire base64 encoded JSON payload. 
    NONE = 1; // Don't forward anything
    // other variants could be added later
}
```

Reply:
SG